bots/psuteam7bot/church.js

```diff
@@ -1,30 +1,107 @@
 import {BCAbstractRobot, SPECS} from 'battlecode';
+import movement from './movement.js';
+import combat from './combat.js';
+import communication from './communication.js';
+//import { setFlagsFromString } from 'v8';
 
 const church = {};
-//Initial starting health of 100 and vision radius of 100
 
-//are we explicitely assigning the unique 32 bit integer id?
-//this unit always occupies single tile
-//what about if the health reduces to zero and the unit is removed? how do we write it here?
+church.UNITTYPE = ["CASTLE", "CHURCH", "PILGRIM", "CRUSADER", "PROPHET" , "PREACHER"]
+
+church.maxKarbonite = SPECS.UNITS[SPECS.CHURCH].KARBONITE_CAPACITY;
+church.maxFuel = SPECS.UNITS[SPECS.CHURCH].FUEL_CAPACITY;
 
 church.doAction = (self) => {
-    self.log("church " + self.id + " taking turn.");
-    return;
+     self.log("church" + self.id + "taking turn.");
+
+    if(self.me.turn == 1){
+        const karboniteDepots = movement.getResourcesInRange(self.me, 36, self.karbonite_map);
+        karboniteDepots.forEach(depot => {
+            self.castleBuildQueue.push({unit: "PILGRIM", x: depot.x, y: depot.y});
+        })
+
+        const fuelDepots = movement.getResourcesInRange(self.me, 16, self.fuel_map)
+        fuelDepots.forEach(depot => {
+            self.castleBuildQueue.push({unit: "PILGRIM", x: depot.x, y: depot.y});
+        })
+
+        const mirrorCastle = movement.getMirrorCastle(self.me, self.map)
+        self.target = mirrorCastle;
+        self.log(self.castleBuildQueue);
+
+        return church.castle.buildFromQueue(self);
```

Comment:
Just `castle.buildFromQueue` I think, You can reuse that method from `castle` just fine